internal/kubernetes/operator/project/project.go

```diff
@@ -437,6 +436,17 @@ func buildIntegrations(intProvider store.IntegrationLister, projectID, targetNam
 	return result, intSecrets, nil
 }
 
+func getIntegrationType(val reflect.Value) string {
+	for i := 0; i < val.NumField(); i++ {
+		field := val.Field(i)
+		if !field.IsNil() {
+			nameField := field.Elem().FieldByName("Type")
+			return *nameField.Interface().(*string)
+		}
+	}
+	return ""
+}
```

Comment:
is this the best way the new sdk has to know the type of an integration? feels convoluted

Reply:
This is the structure:
```
type PaginatedIntegration struct {
    Links      []Link        `json:"links,omitempty"`
    Results    []Integration `json:"results,omitempty"`
    TotalCount *int32        `json:"totalCount,omitempty"`
}
type Integration struct {
    Datadog        *Datadog
    MicrosoftTeams *MicrosoftTeams
    NewRelic       *NewRelic
    OpsGenie       *OpsGenie
    PagerDuty      *PagerDuty
    Prometheus     *Prometheus
    Slack          *Slack
    VictorOps      *VictorOps
    Webhook        *Webhook
}
```

If we can have IntegrationType field then this can be simplified :(

Reply:
Yeaps this works. Just verified. I will update and push the changes.